src/Servers/Kestrel/Core/src/Internal/Http/HttpHeaders.Generated.cs

```diff
@@ -3464,778 +3595,407 @@ protected override bool CopyToFast(KeyValuePair<string, StringValues>[] array, i
-                }
-                if ((_bits & 0x10000000000L) != 0)
-                {
-                    if (arrayIndex == array.Length)
+                    array[arrayIndex] = new KeyValuePair<string, StringValues>("User-Agent", _headers._UserAgent);
+                    ++arrayIndex;
+                }
+                if ((_bits & 0x10000000000L) != 0)
+                {
+                    if (arrayIndex == array.Length)
+                    {
+                        return false;
+                    }
+                    array[arrayIndex] = new KeyValuePair<string, StringValues>("DNT", _headers._DNT);
+                    ++arrayIndex;
+                }
+                if ((_bits & 0x20000000000L) != 0)
+                {
+                    if (arrayIndex == array.Length)
+                    {
+                        return false;
+                    }
+                    array[arrayIndex] = new KeyValuePair<string, StringValues>("Upgrade-Insecure-Requests", _headers._UpgradeInsecureRequests);
+                    ++arrayIndex;
+                }
+                if ((_bits & 0x40000000000L) != 0)
+                {
+                    if (arrayIndex == array.Length)
+                    {
+                        return false;
+                    }
+                    array[arrayIndex] = new KeyValuePair<string, StringValues>("Origin", _headers._Origin);
+                    ++arrayIndex;
+                }
+                if ((_bits & 0x80000000000L) != 0)
+                {
+                    if (arrayIndex == array.Length)
+                    {
+                        return false;
+                    }
+                    array[arrayIndex] = new KeyValuePair<string, StringValues>("Access-Control-Request-Method", _headers._AccessControlRequestMethod);
+                    ++arrayIndex;
+                }
+                if ((_bits & 0x100000000000L) != 0)
+                {
+                    if (arrayIndex == array.Length)
+                    {
+                        return false;
+                    }
+                    array[arrayIndex] = new KeyValuePair<string, StringValues>("Access-Control-Request-Headers", _headers._AccessControlRequestHeaders);
+                    ++arrayIndex;
+                }
+                if (_contentLength.HasValue)
+                {
+                    if (arrayIndex == array.Length)
+                    {
+                        return false;
+                    }
+                    array[arrayIndex] = new KeyValuePair<string, StringValues>("Content-Length", HeaderUtilities.FormatNonNegativeInt64(_contentLength.Value));
+                    ++arrayIndex;
+                }
+            ((ICollection<KeyValuePair<string, StringValues>>)MaybeUnknown)?.CopyTo(array, arrayIndex);
+
+            return true;
+        }
+        
+        [MethodImpl(MethodImplOptions.AggressiveOptimization)]
+        public unsafe void Append(Span<byte> name, Span<byte> value)
```

Comment:
All of this unsafe logic assumes a little-endian machine. I _think_ that's a safe assumption at the moment, and I wouldn't hold up this PR for it, but it should eventually be addressed.

Reply:
Is very safe (currently) as StartAsync will throw if its not:

https://github.com/aspnet/AspNetCore/blob/b44680762ef34c91f5fe0d411ddd8fe0b4e3d06b/src/Servers/Kestrel/Core/src/KestrelServer.cs#L121-L128

Reply:
Thanks for double-checking. :)